parcels/fieldfilebuffer.py

```diff
@@ -568,9 +694,7 @@ def data_access(self):
                         self.chunk_mapping[chunkIndex] = chunkDim
                         chunkIndex += 1
                     self._chunkmap_to_chunksize()
-                    if self.rechunk_callback_fields is not None:
-                        self.rechunk_callback_fields()
-                        self.chunking_finalized = True
+                    self.chunking_finalized = True
```

Comment:
Here, I have my first objection. This will fail if the rechunking (based on the memory budget in the dask config file) mandates more, smaller chunks than the failsafe was able to establish - which is very likely given that the default chunk budget is 128MB and the failsafe in a 32GB system under theoretical conditions is roughly 10 GB. It doesn't fail the tests because, in order to replicate and test for a specific outcome of the chunking, we set this chunk budget manually in the tests (we have to). 

Overall this line says: If `chunksize='auto'" was chosen, then the final chunksize is determined with the very first file access. In order to resize the `load_chunks` array of the related field to the new chunksize, the field needs to update its chunk information. This is done via the callback that has been removed here.

Please restore the previous version for those 3 lines.

Reply:
Well, the problem is that this `self.rechunk_callback_fields()` is called for every new file that is opened, and because the filed.chuck_setup also sets the chunk_info to zero everywhere, the info on which chunks are occupied is lost. This can be a severe slowdown, because it means that Parcles will need to figure out which chunks are occupied and which aren't every time a new file is opened

Perhaps we should fix the if-condition then so that it is only called once per Field, not on every FileBuffer instance?